packages/medusa/src/services/batch-job.ts

```diff
@@ -71,6 +136,200 @@ class BatchJobService extends TransactionBaseService<BatchJobService> {
       }
     )
   }
+
+  async create(data: BatchJobCreateProps): Promise<BatchJob> {
+    return await this.atomicPhase_(async (manager) => {
+      const batchJobRepo: BatchJobRepository = manager.getCustomRepository(
+        this.batchJobRepository_
+      )
+
+      const batchJob = batchJobRepo.create(data)
+      const result = await batchJobRepo.save(batchJob)
+
+      await this.eventBus_
+        .withTransaction(manager)
+        .emit(BatchJobService.Events.CREATED, {
+          id: result.id,
+        })
+
+      return result
+    })
+  }
+
+  async update(
+    batchJobId: string,
+    data: BatchJobUpdateProps
+  ): Promise<BatchJob> {
+    return await this.atomicPhase_(async (manager) => {
+      const batchJobRepo: BatchJobRepository = manager.getCustomRepository(
+        this.batchJobRepository_
+      )
+
+      let batchJob = await this.retrieve(batchJobId)
+
+      const { context, ...rest } = data
+      if (context) {
+        batchJob.context = { ...batchJob.context, ...context }
+      }
+
+      Object.keys(rest)
+        .filter((key) => typeof rest[key] !== `undefined`)
+        .forEach((key) => {
+          batchJob[key] = rest[key]
+        })
+
+      batchJob = await batchJobRepo.save(batchJob)
+
+      await this.eventBus_
+        .withTransaction(manager)
+        .emit(BatchJobService.Events.UPDATED, {
+          id: batchJob.id,
+        })
+
+      return batchJob
+    })
+  }
+
+  async updateStatus(
```

Comment:
thought(non-blocking): should we make this protected don't think this should be exposed in the public API?

Reply:
yes agree, in that case, I should add the other status update such as failed for example